### ISSUE TITLE
Which Linux capabilities are needed by Nethermind to run Docker with only the needed ones?

### ISSUE THREAD

**Issue by luarx** [NethermindEth/nethermind#6566]
**Description**
I am trying to run Nethermind with with limited Linux capabilities using a Docker container

**Actual behavior**
When I try to do that, I see this error 
`Failed to create CoreCLR, HRESULT: 0x8007000E`

**Expected behavior**
I would like to run Nethermind with limited Linux capabiites, could you point out which ones need it, please? 🙏 

**Also I would be good if nethermind runs as a non-root user by default because of security reasons**

**Desktop (please complete the following information):**
Please provide the following information regarding your setup:
 - Version: 1.22.0
 - Installation Method: Docker
 - Consensus Client: lighthouse:v4.4.1

**Logs**
`Failed to create CoreCLR, HRESULT: 0x8007000E`

**Comment by rubo** [comment]
Judging by the error message, it looks like you're not using the official Docker image. Please provide more details on how you run it, with which image, the exact command, etc., so we can diagnose the issue.

**Comment by luarx** [comment]
We are using the official docker image: `nethermind/nethermind:1.22.0` in Kubernetes 👀 
Running it with the default image command and the options that I shared above and trying to limit r/w permissions using `readOnlyRootFilesystem: true`

**Comment by rubo** [comment]
I'm afraid it won't work that way. The error code `0x8007000E` means "out of resources", in this case, file system-related resources. When that error is thrown, Nethermind is not even initialized, as the error comes from the .NET runtime when it's starting up. My guess is something is wrong with mounted volume permissions, or the temp directory is not write-accessible. You may find some workarounds [here](https://github.com/dotnet/runtime/issues/3168).

**Comment by luarx** [comment]
Thanks for pointing out to the root cause @rubo, will check 🙌 

Maybe it is a thing to consider when releasing a Docker image with a non-root user by default ([discussed here](https://github.com/NethermindEth/nethermind/issues/6567)).
**Being able to run a Docker image with a read only root file system is also a security recommendation** 🙏
Wdyt?

**Comment by rubo** [comment]
Yeah, we'll check the possibility for that, but I'm not sure it can be easily achieved.

**Comment by rubo** [comment]
Running Docker with `--read-only` works. But I'm not sure about other approaches.
